lib/elixir/lib/record.ex

```diff
@@ -255,7 +255,15 @@ defmodule Record do
   """
   defmacro defrecord(name, tag \\ nil, kv) do
     quote bind_quoted: [name: name, tag: tag, kv: kv] do
+      if Module.defines?(__MODULE__, {name, 0}, :defmacro) do
+        raise CompileError,
```

Comment:
We can raise a regular ArgumentError here. :) The stacktrace will be correct without having to resort to CompileError.

Reply:
The original error being thrown was a CompileError, so I kept it. Having an ArgumentError makes sense, since we're talking about `name` as an argument to `defrecord`. I'll switch it.